tests/test_lexer.py

```diff
@@ -150,3 +150,30 @@ def test_error(self):
             stream.next()
 
         assert 'SourcePosition(' in repr(excinfo.value)
+
+    def test_error_line_number(self):
+        lg = LexerGenerator()
+        lg.add("NEW_LINE", r"\n")
+        l = lg.build()
+
+        stream = l.lex("\nfail")
+        with raises(LexingError) as excinfo:
+            stream.next()
```

Comment:
Please move this line outside the raises block so it's clear which line raises an exception.